### PR TITLE
Added custom data source and improved security

### PR DESCRIPTION
**Pull request by bsidebottomGoogle**
Added custom datasource for pivoting sheets and improved security with Secrets Manager plus some bug fixes

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -246,18 +265,36 @@ def report():
...
-        to_d = datetime.strptime(data['to_date'], '%Y-%m-%d')
-        event_d = datetime.strptime(data['event_date'], '%Y-%m-%d')
 
-        pre_delta = (event_d - from_d).days
-        post_delta = (to_d - event_d).days
+        try:
+          from_d = df.index.get_loc(data['from_date'])
+          to_d = df.index.get_loc(data['to_date'])
+          event_d = df.index.get_loc(data['event_date'])
+        except KeyError:
```
Comment (peterlafferty) [inline main.py:273]: should there be an error log here?

File: main.py [hunk 2]
```diff
@@ -456,12 +504,12 @@ def _get_gs_data():
   
 def getCsvSettings(csv_data: dict, format: str) -> dict:
   csv_settings = []
-  try:
-    csv_columns, csv_date_column = get_csv_columns(csv_data)
-  except ValueError as e:
-    csv_settings.append('error')
-    csv_settings.append(f'Error - There is an error reading the sheet. Please check format and try again.')
-    return csv_settings
+  #try:
```
Comment (peterlafferty) [inline main.py:507]: is this meant to be commented out?

File: main.py (hunk 1 above)
Comment (bsidebottomGoogle) [reply main.py:273]: No as it is a user error so only to inform the user they have selected a date that does not exist in their data (as I now have enabled them to remove dates).

Comment (peterlafferty) [reply main.py:273]: ok. I think it would be better to log it so that to understand user behaviour

**Review by peterlafferty — APPROVED** [review]

**Pull request merged**